File: app/src/main/java/de/mm/android/longitude/BootReceiver.java
package de.mm.android.longitude;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import de.mm.android.longitude.location.Tracking;
import de.mm.android.longitude.util.PreferenceUtil;

public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = BootReceiver.class.getSimpleName();
    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive");
        if (PreferenceUtil.isUsingApp(context) && PreferenceUtil.isTrackingEnabled(context)) { // restart background tracking
            new Tracking().start(context);
        }
    }
}

File: app/src/main/java/de/mm/android/longitude/BootReceiver.kt
package de.mm.android.longitude

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

import de.mm.android.longitude.location.Tracking
import de.mm.android.longitude.util.PreferenceUtil

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive")
        if (PreferenceUtil.isUsingApp(context) && PreferenceUtil.isTrackingEnabled(context)) {
            Tracking().start(context) // restart background tracking
        }
    }

    companion object {
        private val TAG = BootReceiver::class.java.simpleName
    }
}